files/usr/share/jeos-firstboot/jeos-firstboot-functions

```diff
@@ -0,0 +1,156 @@
+stty_size() {
+	set -- `stty size`; LINES=$1; COLUMNS=$2
+	# stty size can return zero when not ready or
+	# its a serial console
+	if [ "$COLUMNS" = "0" -o "$LINES" = "0" ]; then
+		LINES=24
+		COLUMNS=80
+	fi
+}
+stty_size
+
+result=
+list=
+password=''
+modules=()
+
+let dh_menu=LINES-15
+let dh_text=LINES-5
+
+if pushd "/usr/share/jeos-firstboot/modules" &>/dev/null; then
+        for module in *; do
+                if [ -f "${module}" ] && source "${module}"; then
+                        modules+=("${module}")
+                fi
+        done
+        popd &>/dev/null
+fi
+
+call_module() {
+	local module="$1"
+	local module_func="$2"
+	module_function="${module}_${module_func}"
+	[ "$(type -t "${module_function}")" = "function" ] || return 1
```

Comment:
```suggestion
	[ "$(type -t -- "${module_function}")" = "function" ] || return 1
```

To avoid

```
f32:~ # /usr/sbin/jeos-config -- --help
/usr/share/jeos-firstboot/jeos-firstboot-functions: line 33: type: --: invalid option
type: usage: type [-afptP] name [name ...]
Unknown option '--help'
```

Reply:
so still not enough the OPTIND change?

Reply:
The OPTINID change is fine, now it properly tries to execute `--help` as module. That's not a valid module name though, so it has to be treated specially until it's validated by `type`.